packages/ui-form-field/src/FormFieldLayout/props.ts

```diff
@@ -36,16 +36,35 @@ import type { FormMessage } from '../FormPropTypes'
 
 type FormFieldLayoutOwnProps = {
   label: React.ReactNode
+  /**
+   * the id of the input (to link it to its label for a11y)
+   */
   id?: string
+  /**
+   * the element type to render as
+   */
   as?: AsElementType
+  /**
+   * Array of objects with shape: `{
+   *   text: PropTypes.node,
+   *   type: PropTypes.oneOf(['error', 'hint', 'success', 'screenreader-only'])
+   * }`
+   */
   messages?: FormMessage[]
+  /**
+   * id for the form field messages
+   */
   messagesId?: string
   children?: React.ReactNode
   inline?: boolean
   layout?: 'stacked' | 'inline'
   labelAlign?: 'start' | 'end'
+  vAlign?: 'top' | 'middle' | 'bottom'
```

Comment:
As stated below in the allowedProps comment, I decided to add this prop to the component because it can be passed (and forwarded to Grid), our components pass it too. Also, there is a small chance someone would use this component by itself.